nrf_wifi/fw_if/umac_if/src/tx.c

```diff
@@ -1159,6 +1450,44 @@ static void tx_done_tasklet_fn(unsigned long data)
 }
 #endif /* CONFIG_NRF700X_TX_DONE_WQ_ENABLED */
 
+#ifdef CONFIG_NRF700X_RAWDATA_TX
+enum nrf_wifi_status (nrf_wifi_fmac_rawtx_done_event_process)(
+		      struct nrf_wifi_fmac_dev_ctx *fmac_dev_ctx,
+		      struct nrf_wifi_event_raw_tx_done *config)
+{
+	enum nrf_wifi_status status = NRF_WIFI_STATUS_FAIL;
+	struct nrf_wifi_fmac_dev_ctx_def *def_dev_ctx = wifi_dev_priv(fmac_dev_ctx);
+
+	if (!fmac_dev_ctx) {
+		goto out;
+	}
+
+	if (!config) {
+		nrf_wifi_osal_log_err(fmac_dev_ctx->fpriv->opriv,
+				      "%s: Invalid parameters",
+				      __func__);
+		goto out;
+	}
+
+	nrf_wifi_osal_spinlock_take(fmac_dev_ctx->fpriv->opriv,
+				    def_dev_ctx->tx_config.tx_lock);
+
+	status = tx_done_process(fmac_dev_ctx,
+				 config->desc_num);
+
+	nrf_wifi_osal_spinlock_rel(fmac_dev_ctx->fpriv->opriv,
+				   def_dev_ctx->tx_config.tx_lock);
+
+out:
+	if (status != NRF_WIFI_STATUS_SUCCESS) {
```

Comment:
This is fine for now, but if someone adds code before releasing spin lock then they might miss checking the status, so, I would suggest move this above and use a `goto unlock;`.

Reply:
will check

Reply:
Adding code after invoking txdoneprocess will probably not happen. But, I have made the change to move the print statement above the release of the lock.